### PR TITLE
Fix eku check for BRs

### PR DESCRIPTION
**Pull request by kumarde**
Addresses https://github.com/zmap/zlint/issues/167

### CODE REVIEW

File: util/eku.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package util
```
Comment (dadrian) [inline util/eku.go]: May as well just make this part of `ca.go`. https://github.com/zmap/zlint/blob/master/util/ca.go

File: lints/base.go [hunk 2]
```diff
@@ -97,6 +98,9 @@ func (l *Lint) CheckEffective(c *x509.Certificate) bool {
 // CheckEffective()
 // Execute()
```
Comment (dadrian) [inline lints/base.go]: Can you update the first paragraph of the above comment to describe the additional checks you just added, e.g. lints that are checking for BR compliance are only ran against certs that fall under the purview of the BR's.

**Comment by zakird** [comment]
@titanous can you take a look at this too?

File: util/ca.go [hunk 3]
```diff
@@ -29,3 +29,15 @@ func IsSelfSigned(c *x509.Certificate) bool {
 func IsSubscriberCert(c *x509.Certificate) bool {
 	return !IsCACert(c) && !IsSelfSigned(c)
 }
+
+func IsTestableBRCertificate(cert *x509.Certificate) bool {
```
Comment (titanous) [inline util/ca.go]: Perhaps this should be renamed? It checks if a certificate can be used for ServerAuth, so `IsServerAuthCert` or similar?

Comment (kumarde) [reply util/ca.go]: I can do that.

**Comment by kumarde** [comment]
@titanous i've updated the name.

**Pull request merged**